File: app/src/main/java/com/brainwellnessspa/billingOrderModule/models/PlanListBillingModel.java
package com.brainwellnessspa.billingOrderModule.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class PlanListBillingModel {

        @SerializedName("ResponseData")
        @Expose
        public ResponseData responseData;
        @SerializedName("ResponseCode")
        @Expose
        public String responseCode;
        @SerializedName("ResponseMessage")
        @Expose
        public String responseMessage;
        @SerializedName("ResponseStatus")
        @Expose
        public String responseStatus;

    public ResponseData getResponseData() {
        return responseData;
    }

    public void setResponseData(ResponseData responseData) {
        this.responseData = responseData;
    }

    public String getResponseCode() {
        return responseCode;
    }

    public void setResponseCode(String responseCode) {
        this.responseCode = responseCode;
    }

    public String getResponseMessage() {
        return responseMessage;
    }

    public void setResponseMessage(String responseMessage) {
        this.responseMessage = responseMessage;
    }

    public String getResponseStatus() {
        return responseStatus;
    }

    public void setResponseStatus(String responseStatus) {
        this.responseStatus = responseStatus;
    }

    public class ResponseData {

            @SerializedName("Image")
            @Expose
            public String image;
            @SerializedName("Title")
            @Expose
            public String title;
            @SerializedName("Desc")
            @Expose
            public String desc;
            @SerializedName("Plan")
            @Expose
            public ArrayList<Plan> plan = null;

        public String getImage() {
            return image;
        }

        public void setImage(String image) {
            this.image = image;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }

        public String getDesc() {
            return desc;
        }

        public void setDesc(String desc) {
            this.desc = desc;
        }

        public ArrayList<Plan> getPlan() {
            return plan;
        }

        public void setPlan(ArrayList<Plan> plan) {
            this.plan = plan;
        }

        public class Plan{

                @SerializedName("PlanPosition")
                @Expose
                public String planPosition;
                @SerializedName("PlanID")
                @Expose
                public String planID;
                @SerializedName("PlanAmount")
                @Expose
                public String planAmount;
                @SerializedName("PlanCurrency")
                @Expose
                public String planCurrency;
                @SerializedName("PlanInterval")
                @Expose
                public String planInterval;
                @SerializedName("PlanImage")
                @Expose
                public String planImage;
                @SerializedName("PlanTenure")
                @Expose
                public String planTenure;
                @SerializedName("PlanNextRenewal")
                @Expose
                public String planNextRenewal;
                @SerializedName("SubName")
                @Expose
                public String subName;
                @SerializedName("RecommendedFlag")
                @Expose
                public String recommendedFlag;
                @SerializedName("PlanFlag")
                @Expose
                public String planFlag;
                @SerializedName("PlanFeatures")
                @Expose
                public ArrayList<PlanFeature> planFeatures = null;

                public String getPlanPosition() {
                    return planPosition;
                }

                public void setPlanPosition(String planPosition) {
                    this.planPosition = planPosition;
                }

                public String getPlanID() {
                    return planID;
                }

                public void setPlanID(String planID) {
                    this.planID = planID;
                }

                public String getPlanAmount() {
                    return planAmount;
                }

                public void setPlanAmount(String planAmount) {
                    this.planAmount = planAmount;
                }

                public String getPlanCurrency() {
                    return planCurrency;
                }

                public void setPlanCurrency(String planCurrency) {
                    this.planCurrency = planCurrency;
                }

                public String getPlanInterval() {
                    return planInterval;
                }

                public void setPlanInterval(String planInterval) {
                    this.planInterval = planInterval;
                }

                public String getPlanImage() {
                    return planImage;
                }

                public void setPlanImage(String planImage) {
                    this.planImage = planImage;
                }

                public String getPlanTenure() {
                    return planTenure;
                }

                public void setPlanTenure(String planTenure) {
                    this.planTenure = planTenure;
                }

                public String getPlanNextRenewal() {
                    return planNextRenewal;
                }

                public void setPlanNextRenewal(String planNextRenewal) {
                    this.planNextRenewal = planNextRenewal;
                }

                public String getSubName() {
                    return subName;
                }

                public void setSubName(String subName) {
                    this.subName = subName;
                }

                public String getRecommendedFlag() {
                    return recommendedFlag;
                }

                public void setRecommendedFlag(String recommendedFlag) {
                    this.recommendedFlag = recommendedFlag;
                }

                public String getPlanFlag() {
                    return planFlag;
                }

                public void setPlanFlag(String planFlag) {
                    this.planFlag = planFlag;
                }

                public ArrayList<PlanFeature> getPlanFeatures() {
                    return planFeatures;
                }

                public void setPlanFeatures(ArrayList<PlanFeature> planFeatures) {
                    this.planFeatures = planFeatures;
                }

                public class PlanFeature {

                    @SerializedName("Feature")
                    @Expose
                    public String feature;

                    public String getFeature() {
                        return feature;
                    }

                    public void setFeature(String feature) {
                        this.feature = feature;
                    }
                }
            }
        }

}

File: app/src/main/java/com/brainwellnessspa/billingOrderModule/models/PlanListBillingModel.kt
package com.brainwellnessspa.billingOrderModule.models

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class PlanListBillingModel() : Parcelable{
    @SerializedName("ResponseData")
    @Expose
    var responseData: ResponseData? = null

    @SerializedName("ResponseCode")
    @Expose
    var responseCode: String? = null

    @SerializedName("ResponseMessage")
    @Expose
    var responseMessage: String? = null

    @SerializedName("ResponseStatus")
    @Expose
    var responseStatus: String? = null

    constructor(parcel: Parcel) : this() {
        responseCode = parcel.readString()
        responseMessage = parcel.readString()
        responseStatus = parcel.readString()
    }

    class ResponseData() :  Parcelable{
        @SerializedName("Image")
        @Expose
        var image: String? = null

        @SerializedName("Title")
        @Expose
        var title: String? = null

        @SerializedName("Desc")
        @Expose
        var desc: String? = null

        @SerializedName("Plan")
        @Expose
        var plan: ArrayList<Plan>? = null

        constructor(parcel: Parcel) : this() {
            image = parcel.readString()
            title = parcel.readString()
            desc = parcel.readString()
        }


        class Plan() : Parcelable{
            @SerializedName("PlanPosition")
            @Expose
            var planPosition: String? = null

            @SerializedName("PlanID")
            @Expose
            var planID: String? = null

            @SerializedName("PlanAmount")
            @Expose
            var planAmount = null

            @SerializedName("PlanCurrency")
            @Expose
            var planCurrency: String? = null

            @SerializedName("PlanInterval")
            @Expose
            var planInterval: String? = null

            @SerializedName("PlanImage")
            @Expose
            var planImage: String? = null

            @SerializedName("PlanTenure")
            @Expose
            var planTenure: String? = null

            @SerializedName("PlanNextRenewal")
            @Expose
            var planNextRenewal: String? = null

            @SerializedName("SubName")
            @Expose
            var subName: String? = null

            @SerializedName("RecommendedFlag")
            @Expose
            var recommendedFlag: String? = null

            @SerializedName("PlanFlag")
            @Expose
            var planFlag: String? = null

            @SerializedName("PlanFeatures")
            @Expose
            var planFeatures: List<PlanFeature>? = null

            constructor(parcel: Parcel) : this() {
                planPosition = parcel.readString()
                planID = parcel.readString()
                planCurrency = parcel.readString()
                planInterval = parcel.readString()
                planImage = parcel.readString()
                planTenure = parcel.readString()
                planNextRenewal = parcel.readString()
                subName = parcel.readString()
                recommendedFlag = parcel.readString()
                planFlag = parcel.readString()
            }

            class PlanFeature() : Parcelable{
                @SerializedName("Feature")
                @Expose
                var feature: String? = null

                constructor(parcel: Parcel) : this() {
                    feature = parcel.readString()
                }

                override fun writeToParcel(parcel: Parcel, flags: Int) {
                    parcel.writeString(feature)
                }

                override fun describeContents(): Int {
                    return 0
                }

                companion object CREATOR : Parcelable.Creator<PlanFeature> {
                    override fun createFromParcel(parcel: Parcel): PlanFeature {
                        return PlanFeature(parcel)
                    }

                    override fun newArray(size: Int): Array<PlanFeature?> {
                        return arrayOfNulls(size)
                    }
                }
            }

            override fun writeToParcel(parcel: Parcel, flags: Int) {
                parcel.writeString(planPosition)
                parcel.writeString(planID)
                parcel.writeString(planCurrency)
                parcel.writeString(planInterval)
                parcel.writeString(planImage)
                parcel.writeString(planTenure)
                parcel.writeString(planNextRenewal)
                parcel.writeString(subName)
                parcel.writeString(recommendedFlag)
                parcel.writeString(planFlag)
            }

            override fun describeContents(): Int {
                return 0
            }

            companion object CREATOR : Parcelable.Creator<Plan> {
                override fun createFromParcel(parcel: Parcel): Plan {
                    return Plan(parcel)
                }

                override fun newArray(size: Int): Array<Plan?> {
                    return arrayOfNulls(size)
                }
            }

        }

        override fun writeToParcel(parcel: Parcel, flags: Int) {
            parcel.writeString(image)
            parcel.writeString(title)
            parcel.writeString(desc)
        }

        override fun describeContents(): Int {
            return 0
        }

        companion object CREATOR : Parcelable.Creator<ResponseData> {
            override fun createFromParcel(parcel: Parcel): ResponseData {
                return ResponseData(parcel)
            }

            override fun newArray(size: Int): Array<ResponseData?> {
                return arrayOfNulls(size)
            }
        }
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(responseCode)
        parcel.writeString(responseMessage)
        parcel.writeString(responseStatus)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<PlanListBillingModel> {
        override fun createFromParcel(parcel: Parcel): PlanListBillingModel {
            return PlanListBillingModel(parcel)
        }

        override fun newArray(size: Int): Array<PlanListBillingModel?> {
            return arrayOfNulls(size)
        }
    }
}